[BUG] fixing quotes destroys content of attribute

Given the following html file:
```
<my-component list='["foo", "bar"]'></my-component>
```
with recommended options (including `@html-eslint/quotes`), I get the following warning: 

```
foo.html
  1:16  error  Expected double(") quotes but found single(')  @html-eslint/quotes
```

**Actual behavior:**

When running with '--fix', the following happens with the file:
```
<my-component list="]"""></my-component>
```

**Expected behavior:**
Not sure what exactly to do, there are several options:

If the intended external quote is found within the attribute value:
* do not warn
* warn, but do not fix
* swap quote style (`"`->`'` and `'`->`"`)
  * this would result in `list="['foo', 'bar']"`

--- a/docs/rules/quotes.md
+++ b/docs/rules/quotes.md
@@ -2,6 +2,7 @@
 id: quotes
 title: "@html-eslint/quotes"
 ---
+<!-- prettier-ignore-start -->
 
 ## Enforce consistent quoting attributes with double(`"`) or single(`'`)
 
@@ -20,24 +21,18 @@ This rule has two options
 
 Examples of **incorrect** code for this rule with the default `"double"` option:
 
-<!-- prettier-ignore-start -->
 
 ```html
 <div id='foo'> </div>
 <div id=foo> </div>
 ```
 
-<!-- prettier-ignore-end -->
-
 Examples of **correct** code for this rule with the default `"double"` option:
-
-<!-- prettier-ignore-start -->
 
 ```html
 <div id="foo"> </div>
+<div id='containing "double" quotes'>
 ```
-
-<!-- prettier-ignore-end -->
 
 #### "single"
 
@@ -45,15 +40,18 @@ Examples of **incorrect** code for this rule with the `"single"` option:
 
 ```html
 <div id="foo"></div>
-<div id="foo"></div>
+<div id=foo></div>
 ```
 
 Examples of **correct** code for this rule with the default `"single"` option:
 
 ```html
-<div id="foo"></div>
+<div id='foo'></div>
+<div id="containing 'single' quotes"></div>
 ```
 
 ### Further reading
 
 [MDN - Quoting attributes](https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Code_guidelines/HTML#Quoting_attributes)
+
+<!-- prettier-ignore-end -->

--- a/packages/eslint-plugin/lib/rules/quotes.js
+++ b/packages/eslint-plugin/lib/rules/quotes.js
@@ -96,6 +96,10 @@ module.exports = {
      * @param {AttrNode} attr
      */
     function checkQuotes(attr) {
+      if (attr.value.includes(expectedQuote)) {
+        return;
+      }
+
       const [opening, closing] = getQuotes(attr);
       if (QUOTES_CODES.includes(opening)) {
         if (opening === closing && opening !== expectedQuote) {

--- a/packages/eslint-plugin/tests/rules/quotes.test.js
+++ b/packages/eslint-plugin/tests/rules/quotes.test.js
@@ -14,6 +14,14 @@ ruleTester.run("quotes", rule, {
     {
       code: `<div id = " foo">`,
     },
+    // https://github.com/yeonjuan/html-eslint/issues/95
+    {
+      code: `<div id='"foo"'></div>`,
+    },
+    {
+      code: `<div id="'foo'"></div>`,
+      options: ["single"],
+    },
     {
       code: `
 <div id="

